### PR TITLE
Require node 8 for Javascript tests

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -97,12 +97,12 @@ matrix:
           sudo: required
           compiler: gcc
           node_js:
-            - "6"
+            - "8"
```
Comment (axic) [inline .travis.yml]: Actually this is not needed because we install it ourselves using `nvm`.

**Comment by axic** [comment]
Did somehow zeppelin switched back to and old truffle?

```
Starting our own testrpc instance
writeStringToMemory is deprecated and should not be called! Use stringToUTF8() instead!
Error parsing /tmp/tmp.sdcT8oPN8P/contracts/Bounty.sol: Warning: This is a pre-release compiler version, please do not use it in production.
Compilation failed. See above.
```

**Comment by chriseth** [comment]
This is weird. Does that mean it uses the wrong compiler binary?

**Comment by chriseth** [comment]
I somehow suspect that truffle dynamically loads solc.js over the network if we have node 8, but uses a local copy for node 6. Can that be?

**Comment by axic** [comment]
> I somehow suspect that truffle dynamically loads solc.js over the network if we have node 8, but uses a local copy for node 6. Can that be?

According to truffle, no. It uses a fixed version of `solc-js` and that's it. Currently 0.4.18.

File: .travis.yml [hunk 2]
```diff
@@ -97,12 +97,12 @@ matrix:
           sudo: required
           compiler: gcc
           node_js:
-            - "7"
+            - "8"
```
Comment (axic) [inline .travis.yml:87]: Not sure we should use this plus nvm install. One or the other should be enough.

**Comment by axic** [comment]
Zeppelin fails with:
```
Starting our own testrpc instance
writeStringToMemory is deprecated and should not be called! Use stringToUTF8() instead!
Error parsing /tmp/tmp.axRDH0fXqQ/contracts/Bounty.sol: Warning: This is a pre-release compiler version, please do not use it in production.
Compilation failed. See above.
```

This suggests somehow we did not manage to replace `soljson.js` properly since the `writeStringToMemory` issue should not be displayed.

**Comment by axic** [comment]
This is still failing. 

```
 > zeppelin-solidity@1.8.0 test /tmp/tmp.XNkOEKmOCp
> scripts/test.sh

Starting our own testrpc instance
writeStringToMemory is deprecated and should not be called! Use stringToUTF8() instead!
Error parsing /tmp/tmp.XNkOEKmOCp/contracts/AddressUtils.sol: Warning: This is a pre-release compiler version, please do not use it in production.
Compilation failed. See above.
```

The reason is that zeppelin still uses an old version of truffle which in turn uses and old version of solc-js :(

**Comment by chriseth** [comment]
Asked in the zeppelin slack.

**Comment by spalladino** [comment]
@chriseth @axic we have plans to upgrade to the latest truffle version before the next release. As far as I understand that would solve the issue, right?

See https://github.com/OpenZeppelin/zeppelin-solidity/issues/868

**Comment by axic** [comment]
> we have plans to upgrade to the latest truffle version before the next release. As far as I understand that would solve the issue, right?

I think it should. As long as the solc-js dependency is >=0.4.20 it should be fine.

**Comment by spalladino** [comment]
Indeed, we're now at solidity 0.4.21 since https://github.com/OpenZeppelin/zeppelin-solidity/pull/876. Next release is planned for the week of the 23rd.

**Comment by axic** [comment]
Retriggered the tests and it is working now, woohoo!

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Argh, yeah, let's do this!

**Pull request merged**